### PR TITLE
SR-6717 Builtin types are always be printed with Builtin prefix

### PR DESCRIPTION
**Pull request by JGiola**
ASTPrinter now ignore the Options.FullyQualifiedTypesIfAmbiguous value if the containing module is the builtin one.

As stated in the bug report, I've implemented the test in the description to validate the fix but if someone has a more self-contained example I will swap it in no time.

Resolves [SR-6717](https://bugs.swift.org/browse/SR-6717).

### CODE REVIEW

File: lib/AST/ASTPrinter.cpp [hunk 1]
```diff
@@ -3529,6 +3526,9 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
 
     ModuleDecl *M = D->getDeclContext()->getParentModule();
 
+    if (!Options.FullyQualifiedTypesIfAmbiguous && !M->isBuiltinModule())
```
Comment (jrose-apple) [inline lib/AST/ASTPrinter.cpp]: I think this is equivalent today, but we could go even further by always returning true if `M->isBuiltinModule()` and then testing `Options.FullyQualifiedTypesIfAmbiguous` after that.

File: test/type/builtin_types.swift [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// RUN: %target-typecheck-verify-swift
+
+let x = 0
+_ = x._value == x // expected-error {{referencing operator function '==' on 'BinaryInteger' requires that 'Builtin.Int64' conform to 'BinaryInteger'}}
```
Comment (jrose-apple) [inline test/type/builtin_types.swift]: So, the comment about tests in the Jira was about not relying on the implementation details of Int (the `_value` property). Instead, you can make some other type in this file that uses Builtin.Int64 (or whatever) as a stored property by adding `-parse-stdlib` to the test invocation. (You'll then need to `import Swift` to still get access to the normal standard library.)

**Review by jrose-apple — COMMENTED** [review]
Thanks for picking this up! I have a few comments but I'm glad the solution was this straightforward.

File: test/type/builtin_types.swift (hunk 2 above)
Comment (JGiola) [reply test/type/builtin_types.swift]: No problem

File: lib/AST/ASTPrinter.cpp (hunk 1 above)
Comment (JGiola) [reply lib/AST/ASTPrinter.cpp]: No problem, I've separated the two checks

**Review by jrose-apple — APPROVED** [review]

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
```
17:00:43 
swift/test/type/builtin_types.swift:5:29: error: unexpected error produced: cannot convert value of type 'Int' to specified type 'Builtin.Int64'
17:00:43 let builtin:Builtin.Int64 = 0
17:00:43                             ^
```

Ah, that makes sense: a builtin type doesn't have any of the initialization logic that a normal type does. You can sidestep this problem by using a function that takes a Builtin.Int64 instead.

**Comment by JGiola** [comment]
Ok I’m lost, I don’t know how to use all the builtin stuff and I can’t figure how to write the test :(

**Comment by jrose-apple** [comment]
There's no real documentation on it, so that's not surprising! I was thinking of something like this:

```swift
import Swift

func test(_ builtin: Builtin.Int64) {
  let x = 0
  _ = x == builtin // expected-error {{operator function '==' requires that 'Builtin.Int64' conform to 'BinaryInteger'}}
}
```

**Comment by JGiola** [comment]
Oh yes, I was rethinking on it during work, and it struck me that I was only suppose to trigger an error not generate a valid Builtin value to use in the current check. Sorry for the trouble

**Comment by JGiola** [comment]
Ok I’ve updated the test

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
Looks like it worked! …but other tests need to be updated for this change as well. Mind getting those too?

**Comment by JGiola** [comment]
I'm on it

**Comment by jrose-apple** [comment]
Ah…what about the other failures? https://ci.swift.org/job/swift-PR-Linux/16520/consoleFull

You should be able to run the tests locally as well, either by using build-script or one of the lit.py invocations in docs/Testing.md.

**Comment by JGiola** [comment]
Ops, my bad. I haven’t seen the SILOptimizer one in the log. I’ve fixed them, and I will running the full suite tonight for catching other failures if there are any.

**Comment by JGiola** [comment]
Ok now, at least on a mac, all tests pass!

**Comment by theblixguy** [comment]
@swift-ci please test

**Comment by jrose-apple** [comment]
Thanks, @JGiola!

**Pull request merged**